sky/skylet/providers/aws/node_provider.py

```diff
@@ -281,10 +281,9 @@ def create_node(self, node_config, tags, count) -> Dict[str, Any]:
                     "Name": "tag:{}".format(TAG_RAY_NODE_KIND),
                     "Values": [tags[TAG_RAY_NODE_KIND]],
                 },
-                {
-                    "Name": "tag:{}".format(TAG_RAY_LAUNCH_CONFIG),
-                    "Values": [tags[TAG_RAY_LAUNCH_CONFIG]],
-                },
+                # SkyPilot: removed TAG_RAY_LAUNCH_CONFIG to allow reusing nodes
+                # with different launch configs.
+                # Reference: https://github.com/skypilot-org/skypilot/pull/1671
```

Comment:
Good point! I added the tests in the comment below. I tested it for cluster with autostop set up, where our status refresh will avoid the code to go into this code. 

When the cluster does not have the autostop set up, it will go in to this code if both head nodes (including the leaked one) are stopped, the previous implementation will reuse the wrong VM, due to we could not distinguish the two VMs.

I changed the implementation to consider the VMs with the same launch config first, and then goes to those nodes with different launch config. Wdyt?

Tested: `sky launch -c test-patched-aws` when there are two stopped clusters with the same cluster name. It correctly restart the one with the same `launch_hash`.